### ISSUE TITLE
Feature suggestion: Fail compile on file contents

### ISSUE THREAD

**Issue by HaasJona** [diffplug/spotless#46]
It's very nice that we can automatically reformat code and replace things with spotless, but we also have sometimes problems that aren't easily fixed by replacements (for example missing license headers (no, just adding them automatically doesn't work because we want to list contributors and so on), usage of prohibited/deprecated classes, swear words in comments and so on.

It would be great if we could do something like

```
spotless {
		java {
			customFailure 'test', {
                            // it = file contents
                            if(it.contains("java.util.Scanner") {
                                fail('please use BufferedReader instead')
                            }
			}
		}
```

of course that's a very rough example. The currect "custom" block unfortunately eats all exceptions more or less silently ("Unable to apply step "), else that would be a workaround by just replacing nothing and throwing an exception.

**Comment by nedtwigg** [comment]
Great idea!  The exceptions are actually swallowed not by the custom block but by a higher part of Spotless.  Some of the complex formatters (especially eclipse) can barf on some inputs, and we didn't want a finnicky formatter library to block a user's build.

How about if we let `Error` and its subclasses through, but continue to swallow `Exception`?  That would allow the use of testing and assertion libraries like JUnit in your custom rules, and it means that Spotless would need almost no modifications.

**Comment by HaasJona** [comment]
I think your suggestion sounds very simple and great.

**Comment by nedtwigg** [comment]
Now available in [2.4.0-SNAPSHOT](https://oss.sonatype.org/content/repositories/snapshots/com/diffplug/gradle/spotless/spotless/).  There's an integration test that shows what it looks like in practice [here](https://github.com/diffplug/spotless/blob/master/src/test/java/com/diffplug/gradle/spotless/ErrorShouldRethrow.java).

Any comments before we ship?

**Comment by HaasJona** [comment]
Looks good to me. Thanks for implementing my suggestion.

**Comment by nedtwigg** [comment]
Shipped in 2.4.0.
